### PR TITLE
Updating the old PR.

### PR DESCRIPTION
**Pull request by Ferks-FK**
It is not ready yet, many things need to be done and tested.

### CODE REVIEW

**Comment by Ferks-FK** [comment]
Regarding the latest update that was released, I am having a visual problem that I need help with.
The script checks that the user does not enter a password, so it returns an error, but even when you enter a password, it is called again.
Example:



Note: I know you need to do the password hiding function, but when I implement that, after you enter a password, the script doesn't respond, I believe it must loop infinitely on something.

**Comment by vilhelmprytz** [comment]
Thank you for submitting this PR! The builds are failing, would you mind adjusting your PR so that the builds do not fail? Thanks! :slightly_smiling_face:

**Comment by Ferks-FK** [comment]
> Thank you for submitting this PR! The builds are failing, would you mind adjusting your PR so that the builds do not fail? Thanks! 🙂

I honestly don't understand how this compilation works, can you explain it to me, or give me an example on how to fix it?

**Comment by Linux123123** [comment]
It is not compilation. It is checking for common errors which may lead to bigger problems down the line.
When you click on the information of the build you will see
image
Here the errors are described. All of your errors appear to be related to not used variables. So to fix these problems you probably need to use the variable or delete it's definition.

**Comment by Ferks-FK** [comment]
It requires some testing.

**Review by vilhelmprytz — CHANGES_REQUESTED** [review]
Thank you for submitting this PR! 🙂 

I understand that you have built this as a separate "script", a new option in `install.sh`. This means that you have to reinvent a lot of the functions and processes already occurring in `install-panel.sh`. Would it not be better if support for installing phpMyAdmin was added as an option inside of the already existing installation script(s)?

**Comment by Ferks-FK** [comment]
@vilhelmprytz Take a look at the new modification 😁

File: configs/pma.conf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+server {
```
Comment (Linux123123) [inline configs/pma.conf:1]: Are you sure 2 more `nginx` configs are needed? Can't you just use the pterodactyl one and replace the root / other changes?

Comment (Ferks-FK) [reply configs/pma.conf:1]: I believe it is good to leave this separate, after all, it is being installed in `/var/www/phpmyadmin`, not that it is mandatory, but it is good to separate.

Comment (Ferks-FK) [reply configs/pma.conf:1]: Oops, I understood the question now 🤣
I'll check it out

**Review by vilhelmprytz — COMMENTED** [review]
Shouldn't `install-pma.sh` be removed now since it looks like you've incorporated the installation of phpMyadmin inside of `install-panel.sh`?

**Comment by Ferks-FK** [comment]
> Shouldn't `install-pma.sh` be removed now since it looks like you've incorporated the installation of phpMyadmin inside of `install-panel.sh`?

Yes, I left it there just for a kind of "backup".
Can I remove it?

**Comment by vilhelmprytz** [comment]
> > Shouldn't `install-pma.sh` be removed now since it looks like you've incorporated the installation of phpMyadmin inside of `install-panel.sh`?
> 
> 
> 
> Yes, I left it there just for a kind of "backup".
> 
> Can I remove it?
> 
> 

Yes, you may remove it. The contents will still be available if looking at older commits, so it won't be completely lost.

**Comment by Ferks-FK** [comment]
> > > Shouldn't `install-pma.sh` be removed now since it looks like you've incorporated the installation of phpMyadmin inside of `install-panel.sh`?
> > 
> > 
> > Yes, I left it there just for a kind of "backup".
> > Can I remove it?
> 
> Yes, you may remove it. The contents will still be available if looking at older commits, so it won't be completely lost.

Done 🤠

File: install-panel.sh [hunk 2]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+#shellcheck source=/dev/null
```
Comment (Ferks-FK) [inline install-panel.sh:2]: See: https://github.com/koalaman/shellcheck/wiki/SC1091

File: install-panel.sh [hunk 3]
```diff
@@ -1176,6 +1146,34 @@ main() {
...
 
+  # Login and FQDN for phpmyadmin #
+  if [ "$INSTALL_PHPMYADMIN" == true ]; then
+    while [ -z "$PMA_FQDN" ] || [ "$FQDN" == "$PMA_FQDN" ]; do
+      echo -n "* Enter the FQDN for phpmyadmin (pma.example.com): "
+      read -r PMA_FQDN
+      [ -z "$PMA_FQDN" ] && print_error "FQDN cannot be empty"
+      [ "$FQDN" == "$PMA_FQDN" ] && print_error "The pterodactyl panels domain cannot be the same as PhpMyAdmins!"
+    done
+      if [ "$CONFIGURE_LETSENCRYPT" == true ]; then
```
Comment (Ferks-FK) [inline install-panel.sh:1157]: Added question to generate SSL certificates for the PhpMyAdmin domain

File: install-panel.sh [hunk 4]
```diff
@@ -1209,6 +1207,9 @@ summary() {
   echo "* Configure Let's Encrypt? $CONFIGURE_LETSENCRYPT"
   echo "* Assume SSL? $ASSUME_SSL"
   echo "* Configure PhpMyAdmin? $INSTALL_PHPMYADMIN"
+  [ "$INSTALL_PHPMYADMIN" == true ] && echo "* Hostname/FQDN PhpMyAdmin: $PMA_FQDN"
```
Comment (Ferks-FK) [inline install-panel.sh:1210]: Added PMA summary, if true.

File: install-panel.sh (hunk 2 above)
Comment (Linux123123) [reply install-panel.sh:2]: Not needed, as already ignored at the project level. Check `travis.yml`.

Comment (Ferks-FK) [reply install-panel.sh:2]: Ok.

File: install-panel.sh [hunk 5]
```diff
@@ -804,6 +834,119 @@ letsencrypt() {
   fi
 }
 
+configure_pma() {
```
Comment (Linux123123) [inline install-panel.sh:837]: I think you should also split this function into multiple smaller functions and use the correct indentation of `2` spaces.

File: install.sh [hunk 6]
```diff
@@ -82,6 +82,8 @@ PANEL_LATEST="$GITHUB_BASE_URL/$SCRIPT_VERSION/install-panel.sh"
 
 WINGS_LATEST="$GITHUB_BASE_URL/$SCRIPT_VERSION/install-wings.sh"
 
+PMA_LATEST="$GITHUB_BASE_URL/$SCRIPT_VERSION/install-pma.sh"
```
Comment (Linux123123) [inline install.sh:85]: Also you are still making changes to `install.sh`, when they aren't needed.

File: install-panel.sh (hunk 5 above)
Comment (Ferks-FK) [reply install-panel.sh:837]: Done

File: install-panel.sh [hunk 7]
```diff
@@ -1030,6 +1209,10 @@ summary() {
   echo "* Configure Firewall? $CONFIGURE_FIREWALL"
   echo "* Configure Let's Encrypt? $CONFIGURE_LETSENCRYPT"
   echo "* Assume SSL? $ASSUME_SSL"
+  echo "* Configure PhpMyAdmin? $INSTALL_PHPMYADMIN"
+  [ "$INSTALL_PHPMYADMIN" == true ] && echo "* Hostname/FQDN PhpMyAdmin: $PMA_FQDN"
+  [ "$INSTALL_PHPMYADMIN" == true ] && echo "* PhpMyAdmin Username: $PMA_MYSQL_USER"
+  [ "$INSTALL_PHPMYADMIN" == true ] && echo "* phpmyadmin Password: $PMA_MYSQL_PASSWORD"
```
Comment (Linux123123) [inline install-panel.sh:1215]: Do not actually show the password. It's a security risk.

File: install.sh [hunk 8]
```diff
@@ -109,6 +109,7 @@ while [ "$done" == false ]; do
     "$PANEL_LATEST"
     "$WINGS_LATEST"
     "$PANEL_LATEST;$WINGS_LATEST"
+    "$PMA_LATEST"
```
Comment (Linux123123) [inline install.sh:112]: You left one line in `install.sh`

File: install-panel.sh [hunk 9]
```diff
@@ -49,12 +49,23 @@ fi
 GITHUB_SOURCE="master"
 SCRIPT_RELEASE="canary"
 
+# Main Variables #
+
+#SUPPORT_LINK="https://pterodactyl-installer.se/discord"
```
Comment (Linux123123) [inline install-panel.sh:54]: What is the SUPPORT_LINK for? (even though it's commented).

File: install-panel.sh [hunk 10]
```diff
@@ -117,6 +132,7 @@ valid_email() {
 print_error() {
   COLOR_RED='\033[0;31m'
   COLOR_NC='\033[0m'
+  COLOR_DEFAULT='\e[0m'
```
Comment (Linux123123) [inline install-panel.sh:135]: What is this used for?

File: install-panel.sh [hunk 11]
```diff
@@ -787,8 +813,12 @@ letsencrypt() {
   # Obtain certificate
   certbot --nginx --redirect --no-eff-email --email "$email" -d "$FQDN" || FAILED=true
 
+  if [ "$CONFIGURE_SSL_PMA" == true ]; then
+    certbot --nginx --redirect --no-eff-email --email "$email" -d "$PMA_FQDN" || FAILED=true
+  fi
+
   # Check if it succeded
-  if [ ! -d "/etc/letsencrypt/live/$FQDN/" ] || [ "$FAILED" == true ]; then
+  if [ ! -d "/etc/letsencrypt/live/$FQDN/" ] && [ ! -d "/etc/letsencrypt/live/$PMA_FQDN/" ] || [ "$FAILED" == true ]; then
```
Comment (Linux123123) [inline install-panel.sh:821]: That will not work as it would fail, if the user isn't installing PMA and using letsencrypt.

**Review by Linux123123 — CHANGES_REQUESTED** [review]

**Comment by vilhelmprytz** [comment]
This PR now has conflicts with `master`, do you think you could resolve the conflicts?

**Comment by Linux123123** [comment]
This is very outdated. If any work continues, please open a new PR.

**Pull request closed without merging**